src/sfctl/custom_app.py

```diff
@@ -65,6 +68,167 @@ def validate_app_path(app_path):
             'Invalid path to application directory: {0}'.format(abspath)
         )
 
+def print_progress(current, total, rel_file_path, show_progress):
+    """Display progress for uploading"""
+    if show_progress:
+        print('[{}/{}] files, {}'.format(current, total,
+                                         rel_file_path),
+              file=sys.stderr)
+
+def parse_imagestore_connection_string(cluster_manifest_content): #pylint: disable=invalid-name
+    '''
+    Parses the image store connection string from the cluster manifest.
+    '''
+    try:
+        cluster_manifest = json.loads(cluster_manifest_content)
+        cluster_namespace = {
+            'ClusterManifest': 'http://schemas.microsoft.com/2011/01/fabric'}
+        cluster_manifest_str = str(cluster_manifest['Manifest'])
+        image_store_connection_string = ET.fromstring(
+            cluster_manifest_str).find(
+                'ClusterManifest:FabricSettings', cluster_namespace).find(
+                    ".//*[@Name='Management']").find(
+                        ".//*[@Name='ImageStoreConnectionString']").attrib['Value'] #pylint: disable=line-too-long
+        return image_store_connection_string
+    except AttributeError:
+        pass
+    except TypeError:
+        return None
+
+def get_imagestore_connection_string(sesh, endpoint): #pylint: disable=invalid-name
+    """
+    Obtains cluster manifest through REST API and parses the
+    image store connection string  from the cluster manifest
+    """
+    try:
+        from urllib.parse import urlparse, urlencode, urlunparse
+    except ImportError:
+        from urllib import urlencode
+        from urlparse import urlparse, urlunparse  # pylint: disable=import-error
+
+    try:
+        url_path = '$/GetClusterManifest'
+        url_parsed = list(urlparse(endpoint))
+        url_parsed[2] = url_path
+        url_parsed[4] = urlencode(
+            {'api-version': '3.0-preview'})
+        url = urlunparse(url_parsed)
+
+        manifest_response = sesh.get(url)
+        return parse_imagestore_connection_string(manifest_response.content)
+    except Exception: #pylint: disable=broad-except
```

Comment:
Don't do this, you're going to catch things like keyboard interrupts accidentally.